module/VuFind/src/VuFind/Db/Service/ResourceService.php

```diff
@@ -138,4 +199,73 @@ public function getResourceById($id)
         );
         return $resource;
     }
+
+    /**
+     * Use a record driver to assign metadata to the current row.
+     *
+     * @param \VuFind\RecordDriver\AbstractBase $driver    The record driver
+     * @param \VuFind\Date\Converter            $converter Date converter
+     * @param \VuFind\Db\Entity\Resource        $resource  Resource entity
+     *
+     * @return \VuFind\Db\Entity\Resource
+     */
+    public function assignMetadata(
```

Comment:
It looks like assignMetadata is only used in one other place, so it might be a good next step after this PR is merged to finish migrating the related logic in the UpgradeController so we can remove more code from the Laminas Resource classes.